go/teams/loader2.go

```diff
@@ -133,12 +136,61 @@ func (l *TeamLoader) verifyWriterOrReaderPermissions(ctx context.Context,
 	return l.unimplementedVerificationTODO(ctx, nil)
 }
 
+func (l *TeamLoader) walkUpToAdmin(ctx context.Context, team *keybase1.TeamData, admin SCTeamAdmin) (ret *keybase1.TeamData, err error) {
+	target := admin.TeamID.ToTeamID()
+	for team != nil && !team.Chain.Id.Eq(target) {
```

Comment:
oh yeah, that's a loop :D
so, instead of that could load the ancestor directly and check that the names of the ancestor and this team look like an ancestor relationship.

but now I wonder what happens when someone changes a.b -> a.b2 when a.b.c exists.